### ISSUE TITLE
Change default content type for wrap-stacktrace-web?

### ISSUE THREAD

**Issue by danielcompton** [ring-clojure/ring#327]
The default response format for `wrap-stacktrace-web` is text/html, unless text/javascript is provided in the `Accept` header. When using Ring for an API that responds with JSON (for example), the default error response is a big HTML document which is hard to read.

Web browsers all will set text/html as their most preferred response format, but API clients and curl often won't.

My suggestion is that the behaviour of `ex-response` is changed to reverse the logic: if text/html is specified as an accepted format, send it, otherwise send the plain text version.

```
(defn- ex-response
  "Returns a response showing debugging information about the exception.
  Currently supports delegation to either js or html exception views."
  [req ex]
  (let [accept (get-in req [:headers "accept"])]
    (if (and accept (re-find #"^text/html" accept))
      (html-ex-response ex)
      (js-ex-response ex)
)))
```

What are your thoughts?

Relates to https://github.com/ring-clojure/ring/issues/121.

**Comment by weavejester** [comment]
I'm reluctant to change default behaviour that isn't a bug, and it's hard to see this as one, since clients that don't send a proper `Accept` header shouldn't expect data back in any particular format.

Perhaps an option could be added to prefer the JS format, instead.

**Comment by danielcompton** [comment]
I agree, it's certainly not a bug, just a convenience. However I would expect that 99+% of the time when no content type is specified, plain text would be preferred over HTML (although you could argue that's the whole point of providing an Accept header 😄).

I'm happy to put together a PR to make this an option. Would it also make sense to send this response for text/plain content type accepted? I don't know if there's anything about the text/javascript response that is JS specific?

**Comment by weavejester** [comment]
Perhaps... I'd need to think about the best way to do it. Is it just going to be a boolean option, or something more involved, for instance?

I'm planning on putting out an RC of Ring 1.7.0 in the next few days, so this might not make it into the release, as I need to draw the line somewhere.

**Comment by weavejester** [comment]
Fixed by #378.
